java/gateway-core/src/main/java/com/avaulta/gateway/rules/JsonSchemaFilterUtils.java

```diff
@@ -65,9 +65,79 @@ private Object filterBySchema(String path, JsonNode provisionalOutput, JsonSchem
                 //cases like URLs relative to schema URI are not supported
                 throw new RuntimeException("unsupported ref: " + schema.getRef());
             }
+        } else if (schema.hashOneOf()) {
+            // Get first schema with matches its inner condition.
```

Comment:
Following [oneOf](https://json-schema.org/understanding-json-schema/reference/combining.html#oneof), just a set of children with conditions. The first one which match it, the first one used to get the schema